### PR TITLE
Chapter 6: Modelling User

### PR DESCRIPTION
**Pull request by hazwanizafirah**
ready

### CODE REVIEW

**Comment by hazwanizafirah** [comment]
ready

**Comment by hazwanizafirah** [comment]
ready

**Comment by huongnt-2545** [comment]
commented
This pull request has some conflicted
Steps:
```
# 1. Check out main branch
git checkout main
# 2. Pull the latest from remote to local
git pull origin main
# 3. Checkout this branch
git checkout chapter_6
# 4. Rebase to merge the newest resource from main branch
git rebase main
# 5. Fix conflict
# 6. Rebase continue after fix all conflicted file and re-push on git
git rebase --continue
git add .
git push origin chapter_6 -f
```

**Comment by hazwanizafirah** [comment]
ready

**Comment by hazwanizafirah** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  before_save {self.email = email.downcase}
+
+  validates :name, presence: true
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
```
Comment (anlvq-1058) [inline app/models/user.rb:5]:
```suggestion
  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
  before_save {self.email = email.downcase}

  validates :name, presence: true
```

File: test/models/user_test.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
```
Comment (anlvq-1058) [inline test/models/user_test.rb:1]: Don't need to do the test, you can delete this file

**Review by anlvq-1058 — COMMENTED** [review]
cmt

**Comment by hazwanizafirah** [comment]
ready

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+    VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
```
Comment (anlvq-1058) [inline app/models/user.rb:2]:
```suggestion
  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
```

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+  before_save {self.email = email.downcase}
```
Comment (anlvq-1058) [inline app/models/user.rb:3]:
```suggestion

  before_save {self.email = email.downcase}
  
```

**Review by anlvq-1058 — COMMENTED** [review]
cmt

**Comment by hazwanizafirah** [comment]
ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**